src/sentry/uptime/grouptype.py

```diff
@@ -1,15 +1,312 @@
+    env = detector.config.get("environment", "prod")
+
+    # Received time is the actual time the check was performed.
+    received = datetime.fromtimestamp(result["actual_check_time_ms"] / 1000)
+
+    # XXX(epurkhiser): This can be changed over to using the detector ID in the
+    # future once we're no longer using the ProjectUptimeSubscription.id as a tag.
+    project_subscription = get_project_subscription(detector)
+
+    return {
+        "project_id": detector.project_id,
+        "environment": env,
+        "received": received,
+        "platform": "other",
+        "sdk": None,
+        "tags": {
+            "uptime_rule": str(project_subscription.id),
+        },
+        "contexts": {
+            "trace": {"trace_id": result["trace_id"], "span_id": result.get("span_id")},
+        },
+    }
+
+
+class UptimeDetectorHandler(StatefulDetectorHandler[UptimePacketValue, CheckStatus]):
+    @override
+    @property
+    def thresholds(self) -> DetectorThresholds:
+        return {
+            DetectorPriorityLevel.OK: get_active_recovery_threshold(),
+            DetectorPriorityLevel.HIGH: get_active_failure_threshold(),
+        }
+
+    @override
+    def extract_value(self, data_packet: DataPacket[UptimePacketValue]) -> CheckStatus:
+        return data_packet.packet.check_result["status"]
+
+    @override
+    def build_issue_fingerprint(self, group_key: DetectorGroupKey = None) -> list[str]:
+        return build_fingerprint(self.detector)
+
+    @override
+    def extract_dedupe_value(self, data_packet: DataPacket[UptimePacketValue]) -> int:
+        return int(data_packet.packet.check_result["scheduled_check_time_ms"])
+
+    @override
+    def evaluate(
+        self, data_packet: DataPacket[UptimePacketValue]
+    ) -> dict[DetectorGroupKey, DetectorEvaluationResult] | None:
+        result = super().evaluate(data_packet)
+
+        if not result:
+            return result
+
+        # Uptime does not use stateful detector value grouping
+        evaluation = result[None]
+
+        uptime_subscription = data_packet.packet.subscription
+        metric_tags = data_packet.packet.metric_tags
+
+        detector_issue_creation_enabled = features.has(
+            "organizations:uptime-detector-create-issues",
+            self.detector.project.organization,
+        )
+        issue_creation_flag_enabled = features.has(
+            "organizations:uptime-create-issues",
+            self.detector.project.organization,
+        )
+        restricted_host_provider_ids = options.get(
+            "uptime.restrict-issue-creation-by-hosting-provider-id"
+        )
+        host_provider_id = uptime_subscription.host_provider_id
+        host_provider_enabled = host_provider_id not in restricted_host_provider_ids
+
+        issue_creation_allowed = (
+            detector_issue_creation_enabled
+            and issue_creation_flag_enabled
+            and host_provider_enabled
+        )
```

Comment:
This is where we check a few different feature flags an options to bail out of issue creation, the same way we do in the consumer.